### PR TITLE
DDO-1551 Add Environment Functionality needed for Accelerate Metrics

### PR DESCRIPTION
**Pull request by mflinn-broad**
This PR does a couple different things. 

1. Adds the following  `/environments` API routes using controller logic already implemented by @jyang-broad
     - `GET  /environments`
     - `GET /enivonments/:name`
     - `POST /environments`
2. Adds a serialization layer to the environment controller so that the Environment model is decoupled from the processing of both requests and responses to and from clients.
3. Moves the seeding logic used in integration tests to the functional packages rather than having it in `package tools` This is necessary to avoid import cycles when we go to break up the bit `sherlock_integration_test.go` file.
4. Adds some unit tests which don't depend on an actual DB for for `environments` logic

### CODE REVIEW

**Review by em-may — APPROVED** [review]

File: internal/environments/handlers.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package environments
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+// ErrBadCreateRequest is an error type used when a create servie request fails validation checks
```
Comment (em-may) [inline internal/environments/handlers.go:10]:
```suggestion
// ErrBadCreateRequest is an error type used when a create service request fails validation checks
```
Typo

Comment (jyang-broad) [inline internal/environments/handlers.go]: s/servie/service

**Review by jyang-broad — APPROVED** [review]
Looks sane! I'd like to look into having the api call a  `serialize` on controller objects rather than just having the controller always return serialized objects

File: internal/environments/environments.go [hunk 2]
```diff
@@ -38,22 +38,37 @@ func (environmentController EnvironmentController) DoesEnvironmentExist(name str
...
 
 // ListAll is the public api for listing out all environments tracked by sherlock
-func (environmentController *EnvironmentController) ListAll() ([]Environment, error) {
-	return environmentController.store.listAll()
+func (environmentController *EnvironmentController) ListAll() ([]EnvironmentResponse, error) {
+	environments, err := environmentController.store.listAll()
+	if err != nil {
+		return []EnvironmentResponse{}, err
+	}
+	result := EnvironmentsSerializer{environments}
```
[the remark this answers is not in the document]
Comment (mflinn-broad) [reply internal/environments/environments.go:56]: > Can we create a serialize method
That's what the result.Response method is.. maybe I'm not quite undertanding. I don't really see how this prevents running arbitrary logic with the controller in the future.

[the remark this answers is not in the document]
Comment (mflinn-broad) [reply internal/environments/environments.go]: https://github.com/broadinstitute/sherlock/pull/13/commits/359c34863ffdceb4c423346b8698cfbdd6f971ef

[the remark this answers is not in the document]
Comment (mflinn-broad) [reply internal/environments/environments.go]: Applied this suggestion to all packages. Went and removed a bunch of unnecessary pointers from function signatures while I was at it.

**Pull request merged**